src/main/java/seedu/address/model/TodoList.java

```diff
@@ -0,0 +1,120 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.UniqueTodoList;
+
+/**
+ * Wraps all data at the todo list level.
+ * Duplicates are not allowed (by .isSameTask comparison)
+ */
+public class TodoList implements ReadOnlyTodoList {
+
+    private final UniqueTodoList tasks;
+
+    /*
+     * The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
+     * between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
+     *
+     * Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
+     *   among constructors.
+     */
+    {
+        tasks = new UniqueTodoList();
+    }
+
```

Comment:
I think we should find what other ways we can implement non-static init blocks.

Reply:
Yup, noted.